app/controllers/users_controller.rb

```diff
@@ -0,0 +1,49 @@
+class UsersController < ApplicationController
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    if @user.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def login_form
+    @user = User.new
+  end
+
+  def login
+    username = params[:user][:username]
+    found_user = User.find_by(username: username)
+
+    if found_user
+      session[:user_id] = user.id
```

Comment:
```suggestion
      session[:user_id] = found_user.id
```
This is causing the error when I try to login as an existing user